overlay.d/20rhcos-kdump/usr/lib/dracut/modules.d/50rhcos-kdump/module-setup.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# The latest Dracut release that landed in RHEL 8.3 stopped creating
+# `$systemdsystemunitdir/initrd.target.wants` dir, causing
+# `kdump-capture.service` to not be installed correctly and thus making
+# kdump unable to capture the vmcore during a kernel panic.
+# There is a patch in kexec-tools to address this issue
+# https://bugzilla.redhat.com/show_bug.cgi?id=1907253.
+# For now, we will create this directory ourselves, but we should drop
+# this once the proper kexec-tools patches land.
+
+check() {
+    return 0
+}
+
+depends() {
+    return 0
+}
+
+install() {
+    mkdir -p "$initdir/$systemdsystemunitdir/initrd.target.wants"
```

Comment:
Wouldn't it serve the purpose to do

```suggestion
    mkdir "$initdir/$systemdsystemunitdir/initrd.target.wants"
```

instead of the added `misc-ro` test?

Then the compose will naturally fail if the directory already exists.

Reply:
We'd need to order it after 99kdumpbase though, which... is annoyingly already using `99`. I guess `99kdumpbase-rhcos`?

Reply:
> Then the compose will naturally fail if the directory already exists.

What if another module added the `initrd.target.wants` directory before the patch in `kexec-tools` lands? Maybe it's not a big deal, but would it be better to have the "reminder mechanism" more directly tied to the fix in `kexec-tools`?

Reply:
> We'd need to order it after 99kdumpbase though

Wouldn't we want to order _before_, so the directory exists when the kdump modules are installed?

Reply:
Ahh I see, I thought the failure manifested itself at runtime. But you're saying the dracut module won't even install the service if the directory is missing?

In that case, yes you're right. :+1:

Reply:
Yep, otherwise the very important `kdump-capture.service` won't be installed and enabled in the kdump initramfs, but `kdump.service` (to enable kdump) deceivingly does not fail and a bad kdump initramfs is generated, so when a kernel panic happens and we boot into the kdump initramfs, the vmcore will not be captured.